.github/workflows/build-ci-container.yml

```diff
@@ -0,0 +1,51 @@
+
+name: Build CI Container
+
+permissions:
+  contents: read
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - .github/workflows/build-ci-container.yml
+      - '.github/workflows/containers/github-action-ci/**'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - .github/workflows/build-ci-container.yml
+      - '.github/workflows/containers/github-action-ci/**'
+
+jobs:
+  build-ci-container:
+    if: github.repository_owner == 'llvm'
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+    steps:
+      - name: Write Variables
+        id: vars
+        run: |
+          echo "container-name=ghcr.io/$GITHUB_REPOSITORY_OWNER/ci-ubuntu-22.04" >> $GITHUB_OUTPUT
+
+      - name: Checkout LLVM
+        uses: actions/checkout@v4
+
+      - name: Build Container
+        working-directory: ./.github/workflows/containers/github-action-ci/
+        run: |
+          # TODO: Ideally, we would give each container build a unique name,
+          # but we have limited space in the container registry, so we can't
```

Comment:
Do you have a source for this? I believe this falls under Github packages, which should be free for public packages.

It also seems like they don't have any billing setup for the container registry for even enterprise accounts.

Tagging with the short SHA and then also tagging the image as `latest` would probably be good. There should be a way to just push the tag as well without having to push the image twice. This might be something for a follow up patch though.

Reply:
> Do you have a source for this? I believe this falls under Github packages, which should be free for public packages.
> 
> It also seems like they don't have any billing setup for the container registry for even enterprise accounts.
> 

OK, I missed that the limits only applied to private packages.

> Tagging with the short SHA and then also tagging the image as `latest` would probably be good. There should be a way to just push the tag as well without having to push the image twice. This might be something for a follow up patch though.

I will look into this.